src/rust/engine/fs/src/store.rs

```diff
@@ -1505,6 +1508,29 @@ mod remote {
         )).to_boxed() as BoxFuture<_, _>,
       }
     }
+
+    pub fn list_missing_digests(&self, digests: Vec<Digest>) -> Result<Vec<Digest>, String> {
```

Comment:
Oh, actually, one thing... This probably wants to return a `HashSet<Digest>` rather than a `Vec<Digest>`. Should be a trivial change (though slightly awkward for the tests) :)

(Also, I suspect that now that this is returning a `Result` rather than a `BoxFuture`, you no longer need the type hint on the `collect`).